builder/module-options.nix

```
@@ -0,0 +1,301 @@
+{ config, lib, ... }:
+# TODO: add support for assertions
+let
+  latestRelease = import ../latest-release.nix;
+
+  releaseHashesFile = ../hashes/${config.release}.nix;
+  releaseHashes =
+    if (builtins.pathExists releaseHashesFile)
+    then import releaseHashesFile
+    else builtins.throw "No hashed information found about OpenWRT ${config.release}";
+  hashedTarget = releaseHashes.targets.${config.hardware.target}.${config.hardware.subtarget} or
+    (builtins.throw "No hashed information for OpenWRT ${config.release} found for ${config.hardware.target}/${config.hardware.subtarget}");
+  hashedFeeds = releaseHashes.packages.${config.hardware.arch} or
+    (builtins.throw "No hashed information for OpenWRT ${config.release} about packages found for ${config.hardware.arch} architecture");
+  kmodsFeeds =
+    if (lib.versionAtLeast config.release "24")
+    then releaseHashes.kmods.${config.hardware.target}.${config.hardware.subtarget} or
+      (builtins.throw "No hashed information for OpenWRT ${config.release} about Kmods found for ${config.hardware.target}/${config.hardware.subtarget}")
+    else { };
+
+  defaultSumsFileHash = hashedTarget.sha256;
+
+  # TODO: get rid of `pkgs`
+  profiles = import ../profiles.nix { pkgs = null; inherit (config) release; };
+  detectedProfile = profiles.identifyProfile config.hardware.profile;
+  # TODO: move arch to profiles
+  detectedArch = hashedTarget.packagesArch;
+
+
+  fileSubmodule = lib.types.submodule {
+    options = {
+      target = lib.mkOption {
+        type = lib.types.nonEmptyStr;
+        example = "/etc/uci-defaults/99-custom";
+        description = ''
+          Path within the generated OpenWRT image to put the file
+        '';
+      };
+
+      source = lib.mkOption {
+        type = lib.types.path;
+        example = lib.literalExpression "./uci-script";
+        description = ''
+          Path to the source file or folder.
+        '';
+      };
+    };
+  };
+in
+{
+  # Inspired by https://git.sr.ht/~tomeon/nix-openwrt-imagebuilder-module/tree/6c2a5fc19c5a6fd532b99c69e5b4ce14adfd80fe/item/module.nix
+  options = {
+    release = lib.mkOption {
+      type = lib.types.nonEmptyStr;
+      default = latestRelease;
+      defaultText = lib.literalMD "Value of `latest-release.nix`";
```

```suggestion
      defaultText = "Value of `latest-release.nix`";
```
literalMD doesn't exist anymore in unstable